packages/astro/src/cli/info/index.ts

```diff
@@ -12,31 +15,82 @@ interface InfoOptions {
 	flags: Flags;
 }
 
+export const VERSION_PREFIX_REGEX = /[\^~<>=]/g;
+
+type MinimalPackageJSON = {
+	version: string;
+	dependencies: Record<string, string>;
+}
+
+function getVersion(packageJsonPath: string, dependency?: string): string | undefined {
+	try {
+		const resolvedPath = createRequire(import.meta.url).resolve(packageJsonPath);
+		
+		const packageJSON = JSON.parse(readFileSync(resolvedPath, "utf-8")) as MinimalPackageJSON;
+		
+		if (dependency) {
+			if (!packageJSON.dependencies) return undefined;
+			
+			const depVersion = packageJSON.dependencies[dependency];
+			
+			if (!depVersion) return undefined;
+			
+			const sanitizedDepVersion = depVersion.replaceAll(VERSION_PREFIX_REGEX, "");
```

Comment:
Same problem here, we are reporting the specified version range as the installed version. `^5.2` might install `5.13` and lead us into advising the user to update when it is already up to date.